client/my-sites/domains/domain-management/settings/settings-header.tsx

```diff
@@ -0,0 +1,168 @@
+import { isMappedDomain, resolveDomainStatus } from 'calypso/lib/domains';
+import { type as DomainType } from 'calypso/lib/domains/constants';
+import Breadcrumbs from 'calypso/my-sites/domains/domain-management/components/breadcrumbs';
+import {
+	domainManagementEdit,
+	domainManagementList,
+	isUnderDomainManagementAll,
+} from 'calypso/my-sites/domains/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type {
+	SettingsHeaderProps,
+	SettingsHeaderConnectedProps,
+	SettingsHeaderPassedProps,
+} from './types';
+import type { TranslateResult } from 'i18n-calypso';
+import './style.scss';
+
+const SettingsHeader = ( props: SettingsHeaderProps ) => {
+	const { selectedDomainName, isManagingAllDomains } = props;
+	const { __ } = useI18n();
+	let formattedHeaderText = selectedDomainName;
+	if ( ! selectedDomainName ) {
+		formattedHeaderText = isManagingAllDomains ? __( 'All Domains' ) : __( 'Site Domains' );
+	}
+
+	const renderBreadcrumbs = () => {
+		const { selectedSite, currentRoute, selectedDomainName } = props;
+
+		const previousPath = domainManagementEdit(
+			selectedSite.slug,
+			selectedDomainName,
+			currentRoute
+		);
+
+		const items = [
+			{
+				label: __( 'Domains' ),
+				href: domainManagementList( selectedSite?.slug, selectedDomainName ),
+			},
+			{ label: selectedDomainName },
+		];
+
+		const mobileItem = {
+			label: __( 'Back' ),
+			href: previousPath,
+			showBackArrow: true,
+		};
+
+		return <Breadcrumbs items={ items } mobileItem={ mobileItem } />;
+	};
+
+	const renderCircle = () => (
+		<SVG viewBox="0 0 24 24">
+			<Circle cx="12" cy="12" r="12" />
+		</SVG>
+	);
+
+	const renderSuccessBadge = ( description: TranslateResult, icon?: JSX.Element ) => (
+		<Badge className="settings-header__badge settings-header__badge--success">
+			{ icon ? (
+				<Icon icon={ icon } size={ 14 } />
+			) : (
+				<div className="settings-header__badge-indicator">{ renderCircle() }</div>
+			) }
+			{ description }
+		</Badge>
+	);
+
+	const renderWarningBadge = ( description: TranslateResult ) => (
+		<Badge className="settings-header__badge settings-header__badge--success">
+			<div className="settings-header__badge-indicator">{ renderCircle() }</div>
+			{ description }
+		</Badge>
+	);
+
+	const renderPremiumBadge = () => (
+		<Badge className="settings-header__badge settings-header__badge--premium">
+			{ __( 'Premium domain' ) }
+			<Icon icon={ info } size={ 17 } />
+		</Badge>
+	);
+
+	const renderNeutralBadge = ( description: TranslateResult ) => (
+		<Badge className="settings-header__badge settings-header__badge--neutral">
+			{ description }
+		</Badge>
+	);
+
+	const renderTransferOrMappingBadge = ( type: string ) => {
+		if ( type === DomainType.MAPPED ) {
+			return renderNeutralBadge( __( 'Registered with an external provider' ) );
+		}
+
+		return renderNeutralBadge( __( 'Domain transfer' ) );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Domain Transfer' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-header)